src/access/profiling/parser.py

```diff
@@ -46,6 +50,38 @@ def read(self, stream: str) -> dict:
             dict: profiling data.
         """
 
+    def parse_data_series(self, logs: list[str], varname: str, vars: Iterable) -> xr.Dataset:
+        """Given a list of logs containing profiling data, parse the data and return it as a xarray dataset.
+
+        Args:
+            Logs (list): Logs to parse.
+            varname (str): Name of the variable that changes between logs.
+            vars (Iterable): An iterable returning the value of the variable that changes between logs.
```

Comment:
The docstrings could be fleshed out a bit more - perhaps add an example

Does the `logs` variable refer to the parsed profiling data or the raw data?

Reply:
@micaeljtoliveira Do the logs refer to a list of filenames (as shown in the example), or could the logs be the the string contents of the log files (say with `filename.read()`)? If the former, then it might be better for readability to add a filename qualifier - say something like `logs -> log filenames`

Reply:
Argh, the example is wrong. It shouldn't be a filename, but the content of the file. I'll update the example.

Reply:
I've now updated the example. Let me know if it's clear enough now.